nucypher-core/src/dkg.rs

```diff
@@ -21,6 +109,280 @@ pub enum FerveoVariant {
+        fn as_ref(&self) -> &[u8] {
+            self.0.as_bytes()
+        }
+    }
+
+    impl fmt::Display for RequestPublicKey {
+        /// Format public key information.
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            let bytes = self.as_ref();
+            let bytes = if bytes.len() > 8 { &bytes[..8] } else { bytes };
+            write!(f, "RequestPublicKey: {}", hex::encode(bytes),)
+        }
+    }
+
+    impl<'a> From<&'a RequestSecretKey> for RequestPublicKey {
+        /// Compute corresponding [`RequestPublicKey`].
+        fn from(secret: &'a RequestSecretKey) -> RequestPublicKey {
+            let public_key = PublicKey::from(&secret.0);
+            RequestPublicKey(public_key)
+        }
+    }
+
+    impl<'a> ProtocolObjectInner<'a> for RequestPublicKey {
+        fn version() -> (u16, u16) {
+            (1, 0)
+        }
+
+        fn brand() -> [u8; 4] {
+            *b"TRPk"
+        }
+
+        fn unversioned_to_bytes(&self) -> Box<[u8]> {
+            messagepack_serialize(&self)
+        }
+
+        fn unversioned_from_bytes(
+            minor_version: u16,
+            bytes: &[u8],
+        ) -> Option<Result<Self, String>> {
+            if minor_version == 0 {
+                Some(messagepack_deserialize(bytes))
+            } else {
+                None
+            }
+        }
+    }
+
+    impl<'a> ProtocolObject<'a> for RequestPublicKey {}
+
+    /// A request secret key.
+    #[derive(ZeroizeOnDrop)]
+    pub struct RequestSecretKey(pub(crate) StaticSecret);
+
+    impl RequestSecretKey {
+        /// Perform diffie-hellman
+        pub fn diffie_hellman(&self, their_public_key: &RequestPublicKey) -> RequestSharedSecret {
+            let shared_secret = self.0.diffie_hellman(&their_public_key.0);
+            RequestSharedSecret(shared_secret)
+        }
+
+        /// Create secret key from rng.
+        pub fn random_from_rng<T: RngCore + CryptoRng>(csprng: T) -> Self {
+            let secret_key = StaticSecret::random_from_rng(csprng);
+            Self(secret_key)
+        }
+
+        /// Create random secret key.
+        pub fn random() -> Self {
+            Self::random_from_rng(OsRng)
+        }
+
+        /// Returns a public key corresponding to this secret key.
+        pub fn public_key(&self) -> RequestPublicKey {
+            RequestPublicKey::from(self)
+        }
+    }
+
+    impl fmt::Display for RequestSecretKey {
+        /// Format information above secret key.
+        fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+            write!(f, "RequestSecretKey:...")
+        }
+    }
+
+    pub(crate) fn kdf<S: ArrayLength<u8>>(
+        seed: &[u8],
+        salt: Option<&[u8]>,
+        info: Option<&[u8]>,
+    ) -> SecretBox<GenericArray<u8, S>> {
+        let hk = Hkdf::<Sha256>::new(salt, seed);
+
+        let mut okm = SecretBox::new(GenericArray::<u8, S>::default());
+
+        let def_info = info.unwrap_or(&[]);
```

Comment:
I think this idiom should also yield us `&[]`:

```suggestion
        let def_info = info.unwrap_or_default();
```